pysmt/smtlib/printers.py

```diff
@@ -483,7 +544,7 @@ def walk_bool_constant(self, formula, **kwargs):
             return "false"
 
     def walk_str_constant(self, formula, **kwargs):
-        return '"' + formula.constant_value() + '"'
+        return '"' + formula.constant_value().replace('"', '""') + '"'
```

Comment:
The problem with the old approach is that the constant value did not have the quotation marks escaped at all. E.g.
```
In [1]: from pysmt.shortcuts import String

In [2]: String('asdf"')
Out[2]: "asdf""

In [3]: String('asdf"').to_smtlib()
Out[3]: '"asdf""'
```

The pysmt specification says any double-quotes in the output must be escaped with `""`, see
``` :notes
  "SMT-LIB 2.6 has one escape sequence of its own for string literals. Two
   double quotes are used to represent the double-quote character within 
   a string literal (like the one containing this very note). That escape 
   sequence is at the level of the SMT-LIB frontend of a solver, not at the 
   level of this theory. 
  "```
as in the link posted above.

This escaping was not done before, which is what this change addresses.

Reply:
This was originally done before the regex PR, i guess it snuck in. If you want, we can pull it out and move it into a seperate PR.

Reply:
Yes, please. Separate it into its own PR together with a regression test that does exactly what you show in the example above. This will probably be merged faster than the rest of the PR. Thanks.